sycl/source/detail/stream_impl.cpp

```diff
@@ -61,6 +61,29 @@ size_t stream_impl::get_size() const { return BufferSize_; }
 
 size_t stream_impl::get_max_statement_size() const { return MaxStatementSize_; }
 
+void stream_impl::fill(QueueImplPtr Queue) {
+  auto Q = detail::createSyclObjFromImpl<queue>(Queue);
+  Q.submit([&](handler &cgh) {
+    auto StreamBuf =
+        detail::Scheduler::getInstance().StreamBuffersPool.find(this);
+    assert((StreamBuf !=
+            detail::Scheduler::getInstance().StreamBuffersPool.end()) &&
+           "Stream is unexpectedly not found in pool");
+    auto &FlushBuf = StreamBuf->second->FlushBuf;
+    // Only size of buffer_impl object has been resized.
+    // Value of Range field of FlushBuf instance is still equal to
+    // MaxStatementSize only.
+    size_t FlushBufSize = detail::getSyclObjImpl(FlushBuf)->get_count();
+    auto FlushBufAcc = FlushBuf.get_access<access::mode::read_write,
+                                           access::target::global_buffer>(
+        cgh, range<1>(FlushBufSize), id<1>(0));
+    cgh.codeplay_host_task([=] {
+      char *FlushBufPtr = FlushBufAcc.get_pointer();
+      std::memset(FlushBufPtr, 0, FlushBufAcc.get_size());
```

Comment:
This lambda is executed in a distinct thread. Is there any guarantee that access to this memory is serialized i.e. that there is no race-condition between use of stream and its initialization?

Reply:
I dumped a call graph for several tests and memset kernel is shown as dependency for execution of main thread.

Reply:
Added dependency test.